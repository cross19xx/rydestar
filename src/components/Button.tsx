import React from 'react';
import {
  StyleProp,
  StyleSheet,
  Text,
  TouchableOpacity,
  TouchableOpacityProps,
  ViewStyle,
} from 'react-native';
import { Colors } from '../constants';

interface ButtonProps {
  onPress: TouchableOpacityProps['onPress'];
  title: string;
  type: 'primary' | 'secondary';
  style?: StyleProp<ViewStyle>;
}

export default function Button(props: ButtonProps) {
  const buttonStyle = [];
  buttonStyle.push(props.type === 'primary' ? styles.primaryButton : styles.secondaryButton);
  buttonStyle.push(props.style ? props.style : {});

  const textStyle = props.type === 'primary' ? styles.textPrimary : styles.textSecondary;

  return (
    <TouchableOpacity
      activeOpacity={0.75}
      onPress={props.onPress}
      style={[styles.button, ...buttonStyle]}>
      <Text style={[styles.text, textStyle]}>{props.title}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    width: '100%',
    paddingVertical: 18,
    borderRadius: 8,
  },
  primaryButton: {
    backgroundColor: Colors.PRIMARY,
  },
  secondaryButton: {
    backgroundColor: Colors.ACCENT,
  },
  text: {
    textAlign: 'center',
    fontSize: 16,
  },
  textPrimary: {
    color: Colors.WHITE,
  },
  textSecondary: {
    color: Colors.TEXT_PRIMARY,
  },
});
