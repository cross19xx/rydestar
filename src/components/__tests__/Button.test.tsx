import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';
import snapshotDiff from 'snapshot-diff';
import Button from '../Button';

describe('Component test: Button', () => {
  test('triggers the onPress event when the text is tapped', () => {
    const onPress = jest.fn();
    const text = 'Click me';
    const { getByText } = render(<Button onPress={onPress} title={text} type="primary" />);

    const textElem = getByText(text);
    fireEvent.press(textElem);

    expect(onPress).toHaveBeenCalled();
  });

  test('Renders different styles between primary and secondary', () => {
    const onPress = jest.fn();
    const title = 'Random title';
    expect(
      snapshotDiff(
        <Button onPress={onPress} title={title} type="primary" />,
        <Button onPress={onPress} title={title} type="secondary" />,
      ),
    ).toMatchSnapshot();
  });
});
