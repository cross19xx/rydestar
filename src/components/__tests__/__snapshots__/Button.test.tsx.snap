// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component test: Button Renders different styles between primary and secondary 1`] = `
"Snapshot Diff:
- <Button onPress={[MockFunction]} title=\\"Random title\\" type=\\"primary\\" />
+ <Button onPress={[MockFunction]} title=\\"Random title\\" type=\\"secondary\\" />

@@ -8,11 +8,11 @@
    onResponderTerminate={[Function onResponderTerminate]}
    onResponderTerminationRequest={[Function onResponderTerminationRequest]}
    onStartShouldSetResponder={[Function onStartShouldSetResponder]}
    style={
      Object {
-       \\"backgroundColor\\": \\"#F44336\\",
+       \\"backgroundColor\\": \\"#CCCCCC\\",
        \\"borderRadius\\": 8,
        \\"opacity\\": 1,
        \\"paddingVertical\\": 18,
        \\"width\\": \\"100%\\",
      }
@@ -24,11 +24,11 @@
          Object {
            \\"fontSize\\": 16,
            \\"textAlign\\": \\"center\\",
          },
          Object {
-           \\"color\\": \\"#FFFFFF\\",
+           \\"color\\": \\"#212121\\",
          },
        ]
      }
    >
      Random title"
`;
